### PR TITLE
dev-python/smmap2: A sliding window memory map manager

### PR DESCRIPTION
**Pull request by marbre**
Smmap wraps an interface around mmap and tracks the mapped files as
well as the amount of clients who use it. If the system runs out of
resources, or if a memory limit is reached, it will automatically
unload unused maps to allow continued operation.

This package is nearly a version bumped ebuild of dev-python/smmap,
but the core package's name was changed by upstream.

Required to bump dev-python/gitdb (now renamed to gitdb2).

Bug: https://bugs.gentoo.org/619508

Package-Manager: Portage-2.3.5, Repoman-2.3.1

### CODE REVIEW

**Comment by marbre** [comment]
@mgorny This is one of the deps required for nbdime.

**Comment by mgorny** [comment]
> This package is nearly a version bumped ebuild of dev-python/smmap, but the core package's name was changed by upstream.

Whaat? I guess this makes it parallel-installable with the original, and completely incompatible with any applications that might have used it. Correct?

File: dev-python/smmap2/smmap2-2.0.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+LICENSE="BSD"
+KEYWORDS="~amd64 ~x86"
+SLOT="0"
+IUSE="test"
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		dev-python/nose[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/smmap2/smmap2-2.0.1.ebuild]: I don't think they use nose anywhere in the code. In which case, do not add an unnecessary dependency and run it via unittest instead.

Comment (marbre) [inline dev-python/smmap2/smmap2-2.0.1.ebuild]: https://github.com/gitpython-developers/smmap/blob/master/.travis.yml#L16
https://github.com/gitpython-developers/smmap/blob/09023b1013e081755dca575f62c35eb93ee8065a/setup.py#L55

**Comment by marbre** [comment]
Okay, maybe this is mistakable. The ebuild itself is nearly a copy of the smmap, but the package itself is not as it was renamed. And yes, I think this should make it installable in parallel, this is why I have added it as a new ebuild.

File: dev-python/smmap2/smmap2-2.0.1.ebuild (hunk 1 above)
Comment (mgorny) [inline dev-python/smmap2/smmap2-2.0.1.ebuild]: Yes, I know. But technically I think the tests would work as well when run via:

    "${PYTHON}" -m unittest discover

so why add extra deps? In fact, I will probably try to submit a PR upstream to remove unnecessary nose use.

Comment (mgorny) [inline dev-python/smmap2/smmap2-2.0.1.ebuild]: Hmm, I guess they actually use nose to get the coverage report. However, we don't really care about coverage tbh. But then, I guess nose also gives you parallel runs.

So either switch to unittest, or at least remove the dep on nose-xcover which you don't use.

Comment (marbre) [inline dev-python/smmap2/smmap2-2.0.1.ebuild]: That is what I actually meant by "nearly a version bumped ebuild of dev-python/smmap", as it is a takeover ;) I therefore belive that this dep is also wrong in `dev-python/smmap`. But I agree, will fix this and furthermore patch the setup.py accordingly.

Comment (mgorny) [inline dev-python/smmap2/smmap2-2.0.1.ebuild]: Strictly speaking, you don't have to patch setup.py unless you call `esetup.py test`.

Comment (marbre) [inline dev-python/smmap2/smmap2-2.0.1.ebuild]: Thank you for the advice. I drooped nose-xcover.

**Pull request closed without merging**